src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,50 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def get_slack_channel_ids(organization, user, data, peek=False):
```

Comment:
@iProgramStuff I still find the current function very confusing as it is written as we have different return types based on the inputs. Here's a suggestion of how we can improve it:

- Make a function that will iterate through all the triggers (`get_validated_slack_triggers`), validate the schema, add the validated data to a map where the key is the target identifier and the value is `a_s.validated_data`.
- Make another function that can calls `get_validated_slack_triggers`, takes the output map, and calls `get_target_identifier_display_for_integration` on each validated trigger.
- Make another function that calls `get_validated_slack_triggers` and simply returns a boolean of whether the map is truthy (this replaces the `peek=True` behavior).

Does that make sense? This way each function has a consistent return type and behavior.